### PR TITLE
Codecov fix

### PR DESCRIPTION
**Pull request by albinahlback**
Thanks to @fingolfin for your feedback!

I am trying to excluded all `inlines.c` as they aren't tested in the C tests.

### CODE REVIEW

**Comment by albinahlback** [comment]
@fredrik-johansson Have you installed the Codecov app?

**Comment by fredrik-johansson** [comment]
Yes.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -27,8 +27,8 @@ jobs:
           sudo apt install -y libmpfr-dev
           sudo apt install -y autoconf
           sudo apt install -y libtool-bin
-          sudo apt install -y gcovr
           gcc --version
+          gcov --version
```
Comment (fingolfin) [inline .github/workflows/CI.yml:31]: So is gcovr used after all? But you also remove the code installing it. I am confused :-)

Comment (albinahlback) [reply .github/workflows/CI.yml:31]: Oh, I thought `gcov` and `gcovr` was two different things... What am I supposed to use? I have never done this before :-)

**Comment by fredrik-johansson** [comment]
> I am trying to excluded all inlines.c as they aren't tested in the C tests.

Some of them are. Shouldn't we try to ensure that (most) inline functions are tested instead?

**Comment by albinahlback** [comment]
> Some of them are. Shouldn't we try to ensure that (most) inline functions are tested instead? (Many can be tested indirectly.)

Can you give an example? I believe they shouldn't be if they are tested in C.

**Comment by fredrik-johansson** [comment]
At least with lcov, we get coverage reports for the inline functions directly in the header files while ``inlines.c`` get ignored automatically. For example ``http://flintlib.org/coverage/src/arb.h.gcov.html`` shows coverage for inline functions while there is no ``inlines.c`` listed in http://flintlib.org/coverage/src/arb/index.html.

**Comment by fredrik-johansson** [comment]
I'm still curious why codecov doesn't recognize the inline functions in ``gr.h``.

**Comment by fredrik-johansson** [comment]
BTW, I suppose you want to exclude the test programs too. There's little useful information in a coverage report for https://app.codecov.io/gh/flintlib/flint2/blob/automate_coverage/src/arb/test/t-atan.c where the uncovered code consists precisely of the FAIL blocks...

**Comment by albinahlback** [comment]
> At least with lcov, we get coverage reports for the inline functions directly in the header files while `inlines.c` get ignored automatically. For example http://flintlib.org/coverage/src/arb.h.gcov.html shows coverage for inline functions while there is no `inlines.c` listed in http://flintlib.org/coverage/src/arb/index.html.

Yeah, it does seem like the better option. Do you know how to switch to `lcov`? I was just able to imitate how GAP performs their CI, and it looks to me like they are using `gcov`.

**Comment by albinahlback** [comment]
> BTW, I suppose you want to exclude the test programs too. There's little useful information in a coverage report for https://app.codecov.io/gh/flintlib/flint2/blob/automate_coverage/src/arb/test/t-atan.c where the uncovered code consists precisely of the FAIL blocks...

Yeah, sounds very reasonable. Let me push some exclusions for it.

**Comment by albinahlback** [comment]
And please critique as much as you'd like, this is a new subject for me. Feel free to change anything.

**Comment by albinahlback** [comment]
I will merge this, and then play around with `lcov`. Perhaps the `--capture` flag in `lcov` does something special?

**Comment by albinahlback** [comment]
> However, lcov is to my understanding just a front-end for gcov, so I don't know if the differences we are seeing are due to something else.

Yeah, that is my understanding as well. That's why I didn't bother with `lcov` to begin with. Perhaps we need to check what options we use when using `gcov`.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (fingolfin) [reply .github/workflows/CI.yml:31]: No, all is right, I am just stupid and misread `gcov` :-(. Sorry for the confusion!!!

**Comment by fingolfin** [comment]
Codecov just renders whatever data we give it. As far as I know, lcov and gcov work on the same data, too, so I am rather surprised you get different coverage results. I would first double check that both are really run in identical conditions, i.e., with the same tests executed. E.g. you could *also* run `lcov` as part of the CI, and upload its output as an action artifact, and then compare.

**Pull request merged**